microprofile.ls/org.eclipse.lsp4mp.ls/src/main/resources/org/eclipse/lsp4mp/snippets/mp-metrics.json

```diff
@@ -37,7 +37,8 @@
     "body": [
       "@Gauge(",
       "\tname = \"${1:name}\",",
-      "\tdescription = \"${2:description}\"",
+      "\tdescription = \"${2:description}\",",
+      "\tunit = $3",

```

Comment:
unit is a string, so I think it should be 

`"\tunit = \"${3:unit}\"",`

no?

Reply:
I think for the majority of use cases the MetricUnit constants are used. https://github.com/eclipse/microprofile-metrics/blob/master/api/src/main/java/org/eclipse/microprofile/metrics/MetricUnits.java#L34

Ex. from https://quarkus.io/guides/microprofile-metrics
`@Gauge(name = "highestPrimeNumberSoFar", unit = MetricUnits.NONE, description = "Highest prime number so far.")`

So i didn't want to wrap the placeholder in quotes if they would need to remove them to use the MetricUnit constants. 

I would think we leave it as is, or maybe change it to something like: `"\tunit = ${3:\"unit\"}",`?

Reply:
Snippets should generate code without syntax error, that's why I wonder why it misses quote, but I understand your point of view. So I suggest that you use snippet choice https://microsoft.github.io/language-server-protocol/specification#choice which provides MetricUnits.NONE, and other value of MetricUnits and double quote.

Reply:
Updated to use the choice syntax. I only added a couple options. Not sure if we should cover all possible MetricUnits 
options? Seems like it might be too much.

![image](https://user-images.githubusercontent.com/16768710/92020745-a0c49780-ed26-11ea-92b6-bb6be3f501c2.png)

Reply:
I think it's nice now because snippets generates annotation without syntax error. Thanks @rzgry !